instructionAPI/h/Instruction.h

```diff
@@ -149,21 +138,11 @@ namespace Dyninst { namespace InstructionAPI {
     DYNINST_EXPORT cftConstIter cft_end() const { return m_Successors.end(); }
 
     DYNINST_EXPORT bool operator<(const Instruction& rhs) const {
-      if(m_size < rhs.m_size)
-        return true;
-      if(m_size <= sizeof(m_RawInsn.small_insn)) {
-        return m_RawInsn.small_insn < rhs.m_RawInsn.small_insn;
-      }
-      return memcmp(m_RawInsn.large_insn, rhs.m_RawInsn.large_insn, m_size) < 0;
+      return this->m_RawInsn < rhs.m_RawInsn;
```

Comment:
`ff` should be less than `ff 00`, both of these have the same bytes in the array, so you need an extra clause to check for equality and then compare the sizes:

```suggestion
      return this->m_RawInsn < rhs.m_RawInsn || (this->m_size < rhs.m_size && this->m_RawInsn == rhs.m_RawInsn);
```

Reply:
You mean like you mentioned in your previous comment that I read and proceeded to immediately forget with my goldfishian memory?